src/InvertedIndices.jl

```diff
@@ -8,8 +8,34 @@ struct InvertedIndex{S}
     skip::S
 end
 const Not = InvertedIndex
+
+nonBool2Int(x::Bool) = throw(ArgumentError("invalid index $x of type Bool"))
+nonBool2Int(x::Integer) = Int(x)
+
 # Support easily inverting multiple indices without a temporary array in Not([...])
-InvertedIndex(i₁::Integer, i₂::Integer, iₓ::Integer...) = InvertedIndex(TupleVector((i₁, i₂, iₓ...)))
+function InvertedIndex(i₁::Integer, i₂::Integer, iₓ::Integer...)
+    InvertedIndex(TupleVector((nonBool2Int(i₁), nonBool2Int(i₂), nonBool2Int.(iₓ)...)))
+end
+
+"""
+    GeneralNotMultiIndex(indices)
```

Comment:
Why not just store `indices` in the `skip` field of `InvertedIndex`?

Reply:
Because someone could write `Not(indices)` and we could not distinguish it from `Not(indices...)` and we should be able to do it I think.

Reply:
Maybe to expand on the logic:
1. If user passes multiple integers to `Not(idx1, idx2, idxn...)` this gets converted to `Not(Int[idx1, idx2, idxn...])`
2. If the user passes `Not(idx)` then just `idx` is stored to be negated.
3. This works consistently if `idx` is integer as negating `idx` is the same as negating `Int[idx]` (except if `idx` is `Bool` which is a corner case which I think we should disallow)
4. Now if user passes `Not(idx1, idx2, idxn...)` and the values are not integer we do not want them to be wrapped in a vector like it is done for integers. We have a choice. Either we can wrap them in a `Tuple`: `(idx1, idx2, idxn...)` or in a custom unexported type. I preferred a custom unexported type as then the client package (e.g. DataFrames.jl) can clearly know that multiple positional arguments were passed. Otherwise we would need to start allowing `Not((r"a", r"b"))` as a valid operation equivalent to `Not(r"a", r"b")` (i.e. passing a tuple of operations would be equivalent to passing multiple positional arguments) as it would be not possible to distinguish these two calls on the client libary side.

Reply:
Got it. So basically `Not` only allows a single index (with the special case that multiple integers are automatically converted to negating a single vector index), and `GeneralNotMultiIndex` is for multiple indices? Then maybe call this just `NotMultiIndex`?

Reply:
OK, changed. Your understanding is correct. Just to be precise:

>  with the special case that multiple integers are automatically converted to negating a single vector index

they are automatically converted to negating a single vector index of `Int` (and that is why we have to handle this `Bool` issue)